File: solutions/python/bit-manipulation/missing-number.py
# LeetCode 268 — Missing Number (Easy)
# Category: Bit Manipulation · Approach: XOR
# Time: O(n) | Space: O(1)
# Source: https://leetcode.com/problems/missing-number/

def missing_number(nums: list[int]) -> int:
    missing = len(nums)
    for i, x in enumerate(nums):
        missing ^= i ^ x
    return missing

File: solutions/typescript/bit-manipulation/missing-number.ts
// LeetCode 268 — Missing Number (Easy)
// Category: Bit Manipulation · Approach: XOR
// Time: O(n) | Space: O(1)
// Source: https://leetcode.com/problems/missing-number/

function missingNumber(nums: number[]): number {
  let missing = nums.length;
  for (let i = 0; i < nums.length; i++) {
    missing ^= i ^ nums[i];
  }
  return missing;
}
